app/src/main/java/com/willowtree/vocable/settings/EditKeyboardFragment.kt

```diff
@@ -0,0 +1,230 @@
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentEditKeyboardBinding.inflate(inflater, container, false)
+        keys = resources.getStringArray(R.array.keyboard_keys)
+        arguments?.getParcelable<Phrase>(KEY_PHRASE)?.let {
+            phrase = it
+        }
+
+        populateKeys()
+
+        return binding?.root
+    }
+
+    private fun populateKeys() {
+        keys.withIndex().forEach {
+            with(
+                KeyboardKeyLayoutBinding.inflate(
+                    layoutInflater,
+                    binding?.keyboardKeyHolder,
+                    true
+                ).root as ActionButton
+            ) {
+                text = it.value
+                action = {
+                    //This action mimics sentence capitalization
+                    //Example: "This is what's going on in here. Do you get it? Some letters are capitalized."
+                    val currentText = binding?.keyboardInput?.text?.toString() ?: ""
+                    if (isDefaultTextVisible()) {
+                        binding?.keyboardInput?.text = null
+                        binding?.keyboardInput?.append(text?.toString())
+                    } else if (currentText.endsWith(". ") || currentText.endsWith("? ")) {
+                        binding?.keyboardInput?.append(text?.toString())
+                    } else {
+                        binding?.keyboardInput?.append(text?.toString()?.toLowerCase(Locale.getDefault()))
+                    }
+                }
+            }
+        }
+    }
+
+    private fun isDefaultTextVisible(): Boolean {
+        return binding?.keyboardInput?.text.toString() == getString(R.string.keyboard_select_letters)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding?.backButton?.action = {
+            if (binding?.keyboardInput?.text.toString() == phrase.utterance) {
+                parentFragmentManager
+                .beginTransaction()
+                .replace(R.id.settings_fragment_container, EditPresetsFragment())
+                .addToBackStack(null)
+                .commit()
+            } else {
+                showConfirmationDialog()
+            }
+        }
+
+        binding?.saveButton?.let {
+            it.action = {
+                if (!isDefaultTextVisible()) {
+                    binding?.keyboardInput?.text?.let { text ->
+                        if (text.isNotBlank()) {
+                            phrase.utterance = text.toString()
+                            viewModel.updatePhrase(phrase)
+                        }
+                    }
+                }
+            }
+        }
+
+        binding?.keyboardInput?.setText(phrase.utterance)
+
+        binding?.keyboardClearButton?.let {
```

Comment:
If you're only doing one operation inside the `let` block, you can just drop the `let`. Then it's just:
`binding?.keyboardClearButton?.action = { ... }`
The same goes for other views below.